loopy/kernel/data.py

```diff
@@ -351,20 +352,27 @@ class ArrayArg(ArrayBase, KernelArgument):
             An attribute of :class:`AddressSpace` defining the address
             space in which the array resides.
 
-        .. attribute:: is_output_only
+        .. attribute:: is_output
 
             An instance of :class:`bool`. If set to *True*, recorded to be
```

Comment:
> If set to False, the argument is used to return information to the caller.

Done in 7b477101.

Reply:
> Does is_output == False mean that this variable may not be written by the caller? (I'm leaning yes.)

Decided: yes.

- [ ] Document

Reply:
> Do we catch when a callee calls another function and the is_input/is_output bits don't match?

This would be a part of the post-schedule check. Do we plan to make that a part of this PR?